Tools/AP_Periph/adsb.cpp

```diff
@@ -52,6 +52,15 @@ void AP_Periph_FW::adsb_init(void)
     }
 }
 
+static mavlink_message_t chan_buffer;
+mavlink_message_t* mavlink_get_channel_buffer(uint8_t chan) {
+    return &chan_buffer;
```

Comment:
ignores chan if we have heavy periph

Reply:
> ignores chan if we have heavy periph

This is never going to work well - if at all - on heavy periphs right now, as both the GCS library and the Periph ADSB library are going to be attempting to read from `MAVLINK_COMM0`.  I think it is notionally possible that you could configure the GCS library to not read from the USB port and have things work - 'though you would need to do that configuration via a different serial port.  We're off in la-la land now.

No periph which enables the GCS library (i.e. the heavy periphs) currently instantiates the ADSB library.  Past this PR attempting to do so would cause a compilation failure as there would be two definitions of the two key functions (`mavlink_get_channel_buffer` and `mavlink_get_channel_status`).

https://github.com/ArduPilot/ardupilot/pull/24094 (and the relevant pymavlink PR) will make this rather neater.